test/integration/puppetlabs/trapperkeeper/services/scheduler/scheduler_service_test.clj

```diff
@@ -62,37 +73,185 @@
+  (let [configurations [{}
+                        {:group-id :some-group-identifier}]]
+    (doseq [{:keys [group-id]} configurations]
+      (testing "stop-job lets a job complete but does not run it again"
+        (with-app-with-empty-config app [scheduler-service]
+          (let [service (tk/get-service app :SchedulerService)
+                started (promise)
+                stopped (promise)
+                start-time (atom 0)
+                completed (promise)
+                job (fn []
+                      (reset! start-time (System/currentTimeMillis))
+                      (deliver started nil)
+                      (deref stopped)
+                      (deliver completed nil))
+                interval 10
+                interspaced-fn
+                  (if group-id #(interspaced service interval job group-id)
+                               #(interspaced service interval job))
+                job-id (interspaced-fn)]
+            ; wait for the job to start
+            (deref started)
+            (let [original-start-time @start-time]
+              (testing "the job can be stopped"
+                (is (stop-job service job-id)))
+              (deliver stopped nil)
+              (deref completed)
+              ; wait a bit, ensure the job does not run again
+              (testing "the job should not run again"
+                (Thread/sleep 100)
+                (is (= original-start-time @start-time)))
+              (testing "there should be no other jobs running"
+                (is (= 0 (count @(get-jobs service))))))))))))
+
+(defn guaranteed-start-interval-job
+  ([service interval]
+    (guaranteed-start-interval-job service interval nil))
+
+  ([service interval group-id]
+   (let [started (promise)
+         job (fn []
+               (deliver started nil))
+         result (interspaced service interval job group-id)]
+     (deref started)
+     result)))
+
+; This test has a few race conditions, but unlikely to occur in reality
+(deftest ^:integration test-count-job
+  (testing "count-jobs shows correct number of non-group-id jobs"
+    (with-app-with-empty-config app [scheduler-service]
+      (let [service (tk/get-service app :SchedulerService)
+            interval 10
+            job-0 (guaranteed-start-interval-job service interval)]
+        (is (= 1 (count-jobs service)))
+        (let [job-1 (guaranteed-start-interval-job service interval)]
+          (is (= 2 (count-jobs service))
+          (let [job-2 (guaranteed-start-interval-job service interval)]
+            (is (= 3 (count-jobs service)))
+            ; stop the first job and wait a bit
+            (stop-job service job-0)
+            (Thread/sleep 100)
+            (is (= 2 (count-jobs service)))
+            (stop-job service job-1)
+            (Thread/sleep 100)
+            (is (= 1 (count-jobs service)))
+            (stop-job service job-2)
+            (Thread/sleep 100)
```

Comment:
`stop-job` is synchronous, so these `sleeps` are unnecessary.  Or am I misunderstanding something?

Oh, it's because the jobs might be running when `stop-job` is called, isn't it?  It'd be worth adding a comment about that.